File: 0780-max-chunks-to-make-sorted/0780-max-chunks-to-make-sorted.cs
public class Solution {
    public int MaxChunksToSorted(int[] nums) {
        int n = nums.Length;
        int answer = 0 ; 
        SortedSet<int> set = new SortedSet<int>();
        for(int i = 0 ; i < n ; i++){
            set.Add(nums[i]);
            int lastValue = set.Max();
            if(lastValue == i){
                set.Clear();
                answer++;
            }
        }
         return answer;
    }
}

File: 0780-max-chunks-to-make-sorted/0780-max-chunks-to-make-sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& nums) {
        int NumberOfSplite = 0 ;
        int n = nums.size();
        int ans = 0;
        set<int>s;
        for(int i = 0 ; i <n;i++){
            s.insert(nums[i]);
            if((*s.rbegin()) == i){
               
                ans++;
                s.clear();
            }
        }
        return ans;
    }
};